### DISCUSSION TITLE
Get information from reverse proxy instead with a script

### DISCUSSION THREAD

**Discussion by arunoruto** [umami-software/umami#1459] [General]
Would it be possible to extract the needed data from a log file from a reverse proxy or by using a middleware and let the traffic be recorded by umami? I am using [traefik](https://traefik.io/) and would like to track all of my sites, not only my custom sites where I can inject the script tag. I was wondering if the [access log, created by traefik](https://doc.traefik.io/traefik/observability/access-logs/), could be used/parsed to extract the relevant informations needed for saving the analytics. This would also enable to see if other services/ports are being used, and which containers/domains are accessed. Another option is to use a [middleware](https://doc.traefik.io/traefik/middlewares/http/overview/) to intercept the connection and reroute it back (similar to [authelia](https://www.authelia.com/)).

As for now, the only way to add a script tag would be using the traefik plugin [rewrite body](https://github.com/traefik/plugin-rewritebody). Maybe to find the `<body>` tag and replace it with:
```html
<body>
  <script async defer data-website-id="4fb7fa4c-5b46-438d-94b3-3a8fb9bv2e8b" src="http://your-umami-app.com/umami.js"></script>
```
The problem her is, the `data-website-id`, which will be the same for all websites and we couldn't differentiate it (or so I think?).

**Comment by mikecao** [comment]
You can use the tracker functions directly to send the data you want, https://umami.is/docs/tracker-functions

**Comment by arunoruto** [comment]
@mikecao but that would still mean I need to include the script tag into the website, since I need to call the tracker functions within the website code?

**Comment by mikecao** [comment]
You don't actually have to use the script. Just do POST with the correct payload to your umami `/api/collect` endpoint. See https://umami.is/docs/api.
